tests/Unit/Webhook/Service/UpdateLineItemsServiceTest.php

```diff
@@ -0,0 +1,250 @@
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace OAT\SimpleRoster\Tests\Unit\Webhook\Service;
+
+use DateTimeImmutable;
+use Doctrine\ORM\EntityManagerInterface;
+use OAT\SimpleRoster\Entity\LineItem;
+use OAT\SimpleRoster\Repository\LineItemRepository;
+use OAT\SimpleRoster\WebHook\Service\UpdateLineItemsService;
+use OAT\SimpleRoster\WebHook\UpdateLineItemCollection;
+use OAT\SimpleRoster\WebHook\UpdateLineItemDto;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Psr\Log\LoggerInterface;
+
+class UpdateLineItemsServiceTest extends TestCase
+{
+    /** @var LineItemRepository|MockObject */
+    private $lineItemRepository;
+
+    /** @var EntityManagerInterface|MockObject */
+    private $entityManager;
+
+    /** @var MockObject|LoggerInterface */
+    private $logger;
+
+    /** @var UpdateLineItemsService */
+    private $sut;
+
+    protected function setUp(): void
+    {
+        $this->lineItemRepository = $this->createMock(
+            LineItemRepository::class
+        );
+
+        $this->entityManager = $this->createMock(
+            EntityManagerInterface::class
+        );
+
+        $this->logger = $this->createMock(
+            LoggerInterface::class
+        );
+
+        $this->sut = new UpdateLineItemsService(
+            $this->lineItemRepository,
+            $this->entityManager,
+            $this->logger
+        );
+    }
+
+    public function testItIgnoresUnknownUpdates(): void
+    {
+        $this->lineItemRepository->expects($this->never())->method('findBy');
+        $this->lineItemRepository->expects($this->never())->method('save');
+
+        $updateLineItemCollection = new UpdateLineItemCollection(
+            new UpdateLineItemDto(
+                '52a3de8dd0f270fd193f9f4bff05232f',
+                'Unknown',
+                'https://tao.instance/ontologies/tao.rdf#i5fb5',
+                (new DateTimeImmutable())->setTimestamp(1565602371),
+                'qti-interactions-delivery'
+            )
+        );
+
+        $result = $this->sut->handleUpdates($updateLineItemCollection);
+
+        /** @var UpdateLineItemDto $updateLineItemDto */
+        $updateLineItemDto = $result->getIterator()[0];
+
+        $this->assertSame('ignored', $updateLineItemDto->getStatus());
+    }
+
+    public function testItAcceptsUpdate(): void
+    {
+        $this->lineItemRepository->expects($this->once())
+            ->method('findBy')
+            ->with(
+                [
+                    'slug' => ['qti-interactions-delivery']
+                ]
+            )
+            ->willReturn(
+                [
+                    (new LineItem())
+                        ->setSlug('qti-interactions-delivery')
+                ]
+            );
```

Comment:
```
$this->lineItemRepository
        ->expects(self::once())
        ->method('findBy')
        ->with(['slug' => ['qti-interactions-delivery']])
        ->willReturn([(new LineItem())->setSlug('qti-interactions-delivery')]);
```

5 lines instead of 13

Reply:
What's the gain by doing this? Lines of code? For code, readability is better like it's. there is no sense in what you are saying. if you want to avoid lines, then you contradicting yourself by putting the expects(self::once()) into another line instead of keeping it the same.